doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,388 @@
+
+### Access Mode
+
+In access mode a port possesses the following properties
+* Receives untagged/tagged traffic for a particular Vlan
+* Forwards untagged/tagged traffic for a particular Vlan
+
+### Trunk Mode
+
+In trunk mode a port possesses the following properties
+* Receives untagged traffic for a single Vlan
+* Forwards untagged traffic for a single Vlan
+* Receives tagged traffic for one or more Vlan(s)
+* Forwards tagged traffic for one or more Vlan(s)
+
+### Hybrid Mode
+
+In hybrid mode a port possesses the following properties
+* Receives  multiple tagged & untagged traffic for  one or more Vlan(s)
+* Forwards multiple tagged & untagged traffic for  one or more Vlan(s)
+
+### Sample Topology
+
+![Topology](https://user-images.githubusercontent.com/61490193/208155234-3a04ca08-5ca3-42b3-b679-c1efcacddff2.png)
+
+
+__*Figure 1: Sample Topology*__
+
+The figure shows that 3 different VLANs are configured with 6 other hosts. The switch ports connected to hosts would be configured as access ports. The ports that connect switches together would be configured as trunk ports.
+
+## Architecture Design
+
+The overall SONiC architecture will remain the same and no new sub-modules will be introduced. Changes are made only in the CLI container and Config_DB.
+
+![Architecture](https://user-images.githubusercontent.com/61490193/217232727-811b98a8-ce9c-4227-a94a-1c83a4825a62.png)
+
+
+## High-level Design
+
+In this section we will explain sequence diagrams for the implemented features.
+
+### Switch Port Mode
+
+![SwitchPortMode-PortPortChannel](https://user-images.githubusercontent.com/61490193/207826139-3ff0c41a-69af-4018-bf19-c083e1307bf7.png)
+
+
+__*Figure 2: Sequence Diagram Switch Port mode*__
+
+
+**1.** Check the existing_port_mode.
+
+**2.** Check if existing_port_mode!=routed then set port mode as access and update config_db.
+
+**3.** Check if type==access and existing_mode==trunk then show an error message to remove tagged members first and update config_db. Moreover, switching between modes also works here.
+
+**4.** Check if type==trunk && existing_mode==trunk then update port mode.
+
+**5.** if mode==access or trunk && want to assign IP it will show error to switch mode to routed first for IP assignment.
+
+
+**5.** Check if type== routed then show an error message to remove tagged && untagged vlan members first from port and then set_mode entry as “routed” .
+
+**6.** Update PORT table.
+
+
+__*NOTE: This works same for Adding Switchport Mode on a PortChannel*__
+
+
+### Add Multiple VLANs
+ 
+![MultipleVlans](https://user-images.githubusercontent.com/61490193/208155973-801f99e5-bc79-4e4d-9db4-be13decf6637.png)
+
+
+__*Figure 3: Sequence Diagram for adding Multiple VLANs*__
+
+**1.** The SONiC CLI will call the multiple vlan list parser which will parse multiple vlan(s) in a list and return.
+
+**2.** Check if the VLAN is in the range (2-4094) and whether the vid already exists or not.
```

Comment:
Is this HLD introducing vlan range? if it is do you see any performance challenges programming bulk vlans/range to ASIC?  Please include a test case to validate bulk vlans?

Reply:
Yes, we have introduced the vlan range. No performance challenges will be faced using this. Please, see HLD for test cases. Thanks